src/net.cpp

```diff
@@ -1660,16 +1661,23 @@ void CConnman::SocketHandler()
             }
         }
 
-        InactivityCheck(pnode);
     }
     ReleaseNodeVector(vNodesCopy);
 }
 
 void CConnman::ThreadSocketHandler()
 {
+    int64_t nLastCleanupNodes = 0;
+
     while (!interruptNet)
     {
-        DisconnectNodes();
+        if (GetTimeMillis() - nLastCleanupNodes > 1000) {
+            ForEachNode(AllNodes, [&](CNode* pnode) {
+                InactivityCheck(pnode);
+                DisconnectNodes();
```

Comment:
This should be called outside of `ForEachNode`

Reply:
Oh, right. Fixed.